druid/src/window.rs

```diff
@@ -295,7 +305,17 @@ impl<T: Data> Window<T> {
         }
 
         piet.clear(env.get(crate::theme::WINDOW_BACKGROUND_COLOR));
-        self.paint(piet, data, env);
+        self.paint(piet, invalid_rect, data, env);
+
+        // If commands were submitted during anim frame, ask the handler
+        // to call us back on idle so we can process them in a new event/update pass.
+        if !queue.is_empty() {
+            if let Some(mut handle) = self.handle.get_idle_handle() {
+                handle.schedule_idle(RUN_COMMANDS_TOKEN);
+            } else {
+                log::error!("failed to get idle handle");
+            }
```

Comment:
was this code moved from elsewhere? I'm slightly confused.

Reply:
Sorry, it must have  been a faulty rebase.